### DISCUSSION TITLE
Major v7.0.0 with Appdaemon app support

### DISCUSSION THREAD

**Discussion by mmattel** [mmattel/EET-Solmate#64] [Announcements]
# New Major Release v7

This new release provides a major improvement **Run as Appdaemon App** and a deprecation.

## Breaking Changes

See [breaking.md](https://github.com/mmattel/EET-Solmate/blob/main/breaking.md)

## Changelog

See [changelog.md](https://github.com/mmattel/EET-Solmate/blob/main/changelog.md)

**Comment by mmattel** [comment]
@axcro7 @lukashunziker99 @BetaTobi @josy1024 fyi 😄

**Reply by josy1024** [reply]
@mmattel 
nachdem das API setzen und der Node-Red PID-Regler jetzt zuverlässig laufen anbei Doku & Screenshots, wie ich das setzen der Werte Integriert habe.

https://github.com/josy1024/home-automation-guide/blob/main/node-red/node-red.md

aus den Screenshots ersichtlich wurde um 10 und 12 Uhr Tagsüber auch die Batterie zur Einspeisung angezapft.  

(die PID-Reglerdynamik ist bei mir jetzt 10-190 Watt eingestellt und Regelt auf Bezug von 10 Watt) 
der Wert wird maximal alle 60 Sekunden bei EET API gesetzt.

der Regler kann damit auch umgehen, wenn 60 Watt gebraucht werden aber die Solmate zb 70 Watt liefert.  (der Integralteil darf +- 80 Watt nachregeln)

**Reply by mmattel** [reply]
🚀  **Beeindruckend - Gratulation - tolle Integration!!** 🚀 

Dass die ENV auch MQTT Werte liefert...
Da kann sich Wiener Netze noch was abschauen (dz nur über einen selbst zu installierenden IR Reader...)
Ich nutze ein Qubino 3P Smartmeter nach dem Smartmeter der WN (neue Werte bei 5% Änderung, sehr zuverlässig)

Sehr herzlichen Dank für die Erwähnung 😄 

Hinweis, schau dass Du das Markdown File noch etwas verbessern kannst, zb auch die images über md resizen, des geht recht einfach und erhöht die Lesbarkeit enorm. Beim draufclicken werden die dann eh voll dargestellt... (bin professioneller doc schreiber, da konnte ich nicht anders)

**Reply by josy1024** [reply]
bei EVN brauchts einen "mbus" usb<>serial adater und Entschlüsselungscodes (kann man bei der EVN direkt) beantragen - die serielle Schnittstelle liefert dann alle 6 Sekunden einen aktuellen Datensatz aller Werte (Leistungen, Spannung, Strom) 
Die Technik mit MQTT gabs schon - Ich hab's etwas verfeinert und die HA config dafür gemacht.

Doku: https://github.com/josy1024/home-automation-guide/blob/main/README.md#ger%C3%A4te-server 

Danke für den Tipp - meine Dokus sind eher recht zweckmäßig.

**Comment by schoenis** [comment]
Muss ich zuerst auf v6 upgraden oder kann ich von v5 auf v7 wechseln?

**Reply by mmattel** [reply]
Du solltest direkt wechseln können 😃
Lies dir aber bitte unbedingt die appdaemon und config page durch und auch die Release Notes/Breaking Changes.

Und, den alten crond basierten Weg deaktivieren/löschen bevor du final umsteigst.

**Comment by BetaTobi** [comment]
Hi @mmattel 
Klasse, das klingt gut!

Wie funktioniert das dann konkret?
Ich denke, wir müssen uns dann das HA AppDaemon Addon installieren.
https://community.home-assistant.io/t/home-assistant-community-add-on-appdaemon-4/163259

Korrekt?

Und dann? :-D

Also, sollte jetzt kein Problem sein, ich hab nur noch nicht verstanden, wie ich das Ganze konkret bei mir einbinden muss.
Aber ich hab mich auch noch nicht tiefer damit beschäftigt bisher :)

Viele Grüße

**Reply by mmattel** [reply]
Das einfachste ist, Du liest Dir von oben nach unten das readme durch 😄 
Dort ist alles ganz genau beschrieben. Den alten crond basierten Weg deaktivieren/löschen bevor du final umsteigst. Doppelt sollte das nicht laufen.
Ich hab zwar nur einen Solmate, aber in allen Varianten getestet 😃

**Reply by BetaTobi** [reply]
Wird gemacht :)

**Reply by schoenis** [reply]
Wie lösche ich den alten Weg konkret? Danke!

**Reply by mmattel** [reply]
Migration, ein guter Punkt, muss ich noch in die Doku mit aufnehmen...

Nimm die "alte" [prep-ha.md](https://github.com/mmattel/EET-Solmate/blob/main/docs/prep-ha.md) Dokumentation und arbeite sie verkehrt ab, also von unten nach oben. Wichtig ist, dass die alte Version nicht mehr läuft - kein Parallel Betrieb!

Kopier oder merk Dir sicherheitshalber die `mqtt_` und `eet_` Settings der Konfiguration, die kannst Du dann einfach im neuen `.env_0` einsetzen.

Wenn alles läuft, kannst Du das `solmate` Verzeichnis im Container einfach löschen, die Start-Automation löschen und danach einen HA Reboot machen, dass alles auch wirklich weg ist.

**Reply by mmattel** [reply]
@schoenis gerade einen migration guide erstellt, siehe README 😃

**Reply by schoenis** [reply]
ok danke.
was ist mit "comment the respective filename: key. They are mutual exclusive!" bei der appdaemon Konfig gemeint?

**Reply by schoenis** [reply]
und gleich in der nächsten Zeile: "Add to the existing apps.yaml file the content from the example. This is the file that defines which prograns will be started."
Welches example ist gemeint?

**Reply by mmattel** [reply]
In der Appdaemon Beschreibung steht ganz oben:
`There are some files prepared in the appdaemon_config folder easing the installation and preparation process....` 😄

**Reply by schoenis** [reply]
danke, das hilft mir nur leider nicht weiter...bitte für einen nicht Informatiker übersetzen

**Reply by mmattel** [reply]
@schoenis
Du hast ja `esham` mit git gecloned, wie in der Beschreibung angegeben.
Wenn Du in das Verzeichnis reingehst wohin clone gespeichert hat , vermutlich dein `<homedirectory>/solmate`, dann hast Du jetzt alle Files genauso da wie Du sie hier siehst. Prüfe bitte (siehe in der Beschreibung) dass Du das Tag 7.0.1 ausgechecked hast, das ist dz das letzte. Das bindet Deine Sicht auf alles was im Tag eingefroren wurde und nicht auf das wenn Änderungen kommen.
Gehe jetzt, in das geclonte Verzeichnis `appdaemon_config`. Dort findest Du Beispielfiles. Schau Sie Dir einfach mal an und vergleiche Sie mit denen Die Du findest wenn Du auf das Verzeichnis gehst wo Du Deinen HA mit Samba gemountet hast. (Ich denke mal Du hast Windows, im Explorer oben in Die Zeile eingeben: `\\<url-vom-ha`, aber ohne http(s) und ohne port). Jetzt hast Du zwei Fenster und kannst einfach vergleichen bzw Daten von lokal nach dem mount kopieren oder editieren. Hinweis für das Editieren, schau dass Du im richtigen Fenster bist... 😅

**Comment by andreasgatt** [comment]
kann das sein das die .env-sample datei nicht gecloned wird?

**Reply by andreasgatt** [reply]
Also hier auf dem Punkt -

As a starting point, make a copy of the .env-sample file and name it the following:

For Plain Python installations name it .env
For an Appdeamon installation name it .env_0

Wo bekomme ich .env-sample her

**Reply by mmattel** [reply]
@andreasgatt 👋
ich gehe mal davon aus dass du mit git clone wie beschrieben gecloned hast und Windows benutzt. Denn in dem Fall musst Du einfach im Dateiexplorer die Sichtbarkeit von versteckten Dateien einschalten. Das sind alle die mit einem Punkt anfangen. Ich werde das noch in der Doku vermerken.

**Reply by andreasgatt** [reply]
Mac - ja das macht Sinn Danke dir

**Comment by schoenis** [comment]
Bei set boost wattage kann man leider nur mehr 500 W maximal eingeben. 
Die EET app erlaubt bis 800W. 
Kann ich dieses Maximum irgendwie ändern?
Danke!

**Reply by mmattel** [reply]
Klar, schau in die .env-sample, dort findest du alle möglichen Settings die Du verwenden kannst. Das was Du brauchst einfach in die .env übertragen und den Wert einstellen. Danach Appdaemon restarten und das Setting ist aktiv.

**Comment by axcro7** [comment]
Hi @mmattel,
Ich bin noch auf v4.1.0 unterwegs und habe mich bisher darum gedrückt upzudaten (hat seither ohne Probleme funktioniert - never change a running system 😉)
Scheinbar fehlen nach dem Dezember Update von HA eine wichtige Komponenten - siehe die letzten Zeilen der `nohup.log`:
```nohup.log
Once a day queries called by scheduler.
^MShutting down MQTT gracefully.
^MTerminated on request.
Traceback (most recent call last):
  File "/config/shell/solmate/solmate.py", line 7, in <module>
    import schedule
ModuleNotFoundError: No module named 'schedule'
Traceback (most recent call last):
  File "/config/shell/solmate/solmate.py", line 7, in <module>
    import schedule
ModuleNotFoundError: No module named 'schedule'
```

Wenn ich das richtig verstanden habe, könnte ich jetzt direkt auf 7.2.2 updaten? 
Verliere ich beim Löschen der Entitäten auch die ganzen historischen Daten, bzw. lässt sich die History mit den neuen Entitäten vereinen?
Sorry, ich blicke da irgendwie nicht mehr durch 🙈

**Comment by mmattel** [comment]
Hallo @axcro7 👋 

ich arbeite bei der Versionierung ja nach dem [semver](https://semver.org) Verfahren.
Du bist 3 major Versionen hinten, und major bedeutet breaking changes...

Der Umstieg auf Appdaemon bringt viele Vorteile, war aber auch ein breaking change. Die vorherige Installationsvariante (außer Plain) unterstütze ich deshalb nicht mehr. Mein esham läuft auch unter AD und ist mega stabil.

EET bereitet grad eine neue, verbesserte API vor die das Leben der Entwickler leichter machen soll (!) und das kommt nur mit v7 als Basis...

Ich hab mich mal versucht schlau zu machen wie man die History von Entitäten die dann mit einem neuen Namen kommen übernehmen kann, bin aber auf nichts gestoßen das click-und-erledigt geht. Als Ansatz würde ich Dir [Migrate energy statistics from one entity to another](https://community.home-assistant.io/t/migrate-energy-statistics-from-one-entity-to-another/405863) oder [Replace entities and keep history (BREAKING CHANGE starting HA Core 2023.4)](https://community.home-assistant.io/t/replace-entities-and-keep-history-breaking-change-starting-ha-core-2023-4/639465) empfehlen, das ist aber schon gehoben...

Wichtig ist, es gehen mit dem v7 esham NUR die neuen Entity Namen !!!
Die sind dafür aber sicher incl History, auch wenn Du zb Deinen Solmate wegen Totalschaden austauschen musst (neue SN#).

Des weiteren empfehle ich Dir dringend auf v7 umzusteigen, auch wenn das mal am Anfang etwas Arbeit bedeutet. Alles unter v7 ist unsupported. Du bekommst mit v7 auch ein deutlich besseres esham Logging und handling mit. zB kannst Du esham nun per Switch ein- und ausschalten oder mehrere Solmates parallel betreiben, viele Einstellungen lassen sich nun auch von HA betreiben incl Reboot etc.

**Reply by axcro7** [reply]
Das mit den breaking changes hatte ich schon gelesen. Mich würde vor allem interessieren, ob ich Zwischenschritte machen muss, oder direkt auf v7 gehen kann?

**Reply by mmattel** [reply]
v4 war eine ganz andere Installationsart als v7 welche auf AppDaemon basiert. Es ist also wie eine Neuinstallation mit Übernahme von env Settings.
